packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -8,6 +8,7 @@ import 'dart:convert';
 import 'dart:developer' as developer;
 import 'dart:ui' as ui;
 
+import 'package:collection/collection.dart';
```

Comment:
we try to avoid depending on other packages in this repo. ideally we should remove all the dependencies on package:collection rather than adding more.

Reply:
sounds good to me. In that case, I don't think this change worth making

Reply:
Hmm, then what about adding the `firstWhereOrNull` as a (private) extension method?

```
extension<T> on List<T> {
  firstWhereOrNull() => ...;
}
```

By doing so, IMHO the codebase is still a bit neater, since (1) `collection` package is well-known so people feel familiar when reading `firstWhereOrNull` and knows what it is (2) the `firstWhereOrNull` name looks a bit cleaner than `.cast().where(... orElse: null)` and introduce less cognitive overhead.

It is just like extracting the common parts (repeated 10 times in this file) into an (extension) function.